region/brinstar/kraid/Kraid Room.json

```diff
@@ -231,18 +231,56 @@
     {
       "id": 2,
       "link": [1, 1],
-      "name": "Leave Shinecharged",
+      "name": "Leave Shinecharged (Kraid Defeated)",
       "requires": [
         {"canShineCharge": {"usedTiles": 22, "openEnd": 0}},
         "canShinechargeMovement",
         "f_DefeatedKraid",
-        {"shineChargeFrames": 45}
+        {"shineChargeFrames": 40}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {}
+      },
+      "flashSuitChecked": true
+    },
+    {
+      "link": [1, 1],
+      "name": "Leave Shinecharged (X-Mode or Kraid Defeated)",
+      "requires": [
+        "canXMode",
+        "h_XModeThornHit",
+        "h_XModeThornHit",
+        "h_shinechargeMaxRunway",
+        {"or": [
+          {"canShineCharge": {"usedTiles": 22, "openEnd": 0}},
+          {"and": [
+            {"not": "f_DefeatedKraid"},
+            "canRiskPermanentLossOfAccess"
+          ]}
+        ]},
+        "canShinechargeMovement",
+        {"shineChargeFrames": 55},
+        {"or": [
+          {"and": [
+            "canNeutralDamageBoost",
+            "canBeVeryPatient"
+          ]},
+          {"shineChargeFrames": 10}
+        ]}
```

Comment:
I don't follow this `or`, but the rest looks good

Reply:
That `or` just makes it expect 10 more shinecharge frames for the case where you don't do the neutral damage boost to get up to the ledge quicker. My thinking is the damage boost version fits more in Insane, and the 5 lenience shinecharge frames in Extreme isn't enough to make up the difference so you want a little extra lenience here.